newdle/client/src/components/answer/AnswerPage.js

```diff
@@ -18,19 +44,85 @@ export default function AnswerPage() {
   return (
     <div>
       <Grid container>
+        {error && (
+          <Grid.Row centered>
+            <Message error>
+              <p>Something when wrong while sending your answer:</p>
+              <code>{error}</code>
+            </Message>
+          </Grid.Row>
+        )}
+        {saved && (
+          <Grid.Row centered>
+            <Message success>
+              <p>Your answer has been saved!</p>
+            </Message>
+          </Grid.Row>
+        )}
+        {!participantCode && (
+          <Grid.Row>
+            <div className={styles['participant-name-box']}>
+              <h3>Who are you?</h3>
+              <Input
+                autoFocus
+                transparent
+                className={styles['participant-name-input']}
+                placeholder="Please enter your name..."
+                disabled={submitting}
+                onChange={(_, data) => setName(data.value)}
+                onKeyDown={e => {
+                  if (e.key === 'Enter' && canSubmit) {
+                    answerNewdle();
+                  }
+                }}
+              />
+            </div>
+          </Grid.Row>
+        )}
         <Grid.Row columns={2}>
           <Grid.Column width={5}>
             <MonthCalendar />
-            <Header as="h3" className={styles['options-msg']}>
-              {numberOfAvailable} out of {numberOfTimeslots} options chosen
-            </Header>
-            <Checkbox toggle label="Accept all options where I'm available" />
+            <Segment attached="bottom" secondary>
+              <Checkbox toggle label="Accept all options where I'm available" />
+            </Segment>
           </Grid.Column>
           <Grid.Column width={11}>
-            <Calendar />
+            <Calendar getAvailability={!!participantCode} />
           </Grid.Column>
         </Grid.Row>
+        <Grid.Row className={styles['bottom-row']}>
+          <span className={`${styles['options-msg']} ${numberOfAvailable ? '' : 'none'}`}>
+            {numberOfAvailable ? (
+              <>
+                {numberOfAvailable} out of {numberOfTimeslots} options chosen
+              </>
+            ) : (
+              <em>No options chosen</em>
+            )}
+          </span>
+          <Button
+            size="large"
+            color="violet"
+            content="Send you answer"
+            disabled={saved || submitting || !canSubmit}
+            loading={submitting}
+            icon="send"
+            onClick={() => {
+              if (canSubmit) {
+                answerNewdle();
+              }
+            }}
+          />
+        </Grid.Row>
       </Grid>
     </div>
   );
 }
+
+AnswerPage.propTypes = {
```

Comment:
If you use react-router's `useParams()` hooks to access the route params you have a good excuse to not bother with proptypes here :D